src/darwin/frida-helper-backend-glue.m

```diff
@@ -2470,6 +2456,305 @@ static gboolean frida_agent_context_init_functions (FridaAgentContext * self, Gu
+
+  image = gum_darwin_read (task, base, 4096, NULL);
+  header = image;
+
+  if (header->magic == MH_MAGIC)
+    command = image + sizeof (struct mach_header);
+  else
+    command = image + sizeof (struct mach_header_64);
+
+  slide = frida_get_module_slide (command, header->ncmds, base);
+
+  for (command_index = 0;
+       command_index != header->ncmds && initializer == 0;
+       command_index++)
+  {
+    const struct load_command * lc = command;
+
+    if (lc->cmd == LC_SEGMENT || lc->cmd == LC_SEGMENT_64)
+    {
+      gconstpointer sections;
+      gsize section_count, section_index;
+
+      if (lc->cmd == LC_SEGMENT)
+      {
+          const struct segment_command * sc = command;
+
+          sections = sc + 1;
+          section_count = sc->nsects;
+
+          if (strcmp (sc->segname, "__DATA_CONST") != 0 &&
+              strcmp (sc->segname, "__DATA") != 0)
+          {
+            command += lc->cmdsize;
+            continue;
+          }
+      }
+      else
+      {
+          const struct segment_command_64 * sc = command;
+
+          sections = sc + 1;
+          section_count = sc->nsects;
+
+          if (strcmp (sc->segname, "__DATA_CONST") != 0 &&
+              strcmp (sc->segname, "__DATA") != 0)
+          {
+            command += lc->cmdsize;
+            continue;
+          }
+      }
+
+      for (section_index = 0;
+          section_index != section_count && initializer == 0;
+          section_index++)
+      {
+        GumAddress addr;
+        const char * sectname;
+        gsize sectsize;
+
+        if (lc->cmd == LC_SEGMENT)
+        {
+          const struct section * s = sections + (section_index * sizeof (struct section));
+
+          sectname = s->sectname;
+          addr = s->addr + (guint32) slide;
+          sectsize = s->size;
+        }
+        else
+        {
+          const struct section_64 * s = sections + (section_index * sizeof (struct section_64));
+
+          sectname = s->sectname;
+          addr = s->addr + (guint64) slide;
+          sectsize = s->size;
+        }
+
+        if (strcmp (sectname, "__mod_init_func") == 0)
+        {
+          if (lc->cmd == LC_SEGMENT)
+          {
+            guint32 * init_func;
+            g_assert_cmpint (sectsize, ==, 4);
+            init_func = (guint32*) gum_darwin_read (task, addr, sizeof (guint32), NULL);
+            initializer = *init_func;
+            g_free (init_func);
+          }
+          else
+          {
+            guint64 * init_func;
+            g_assert_cmpint (sectsize, ==, 8);
+            init_func = (guint64*) gum_darwin_read (task, addr, sizeof (guint64), NULL);
+            initializer = *init_func;
+            g_free (init_func);
+          }
+        }
+      }
+    }
+
+    command += lc->cmdsize;
+  }
+
+  g_free (image);
+
+  return initializer;
+}
+
+static GumAddress
+frida_get_module_slide (gconstpointer command, gsize ncmds, GumAddress base)
+{
+  gsize command_index;
+  gsize slide = 0;
```

Comment:
As this is the result/return value let's move this variable declaration up one line so it's first.